### PR TITLE
Add checkbox to push artifacts for failed build

### PR DESCRIPTION
**Pull request by AjeetGupta**

### CODE REVIEW

File: src/main/java/com/urbancode/jenkins/plugins/ucdeploy/UCDeployPublisher.java [hunk 1]
```diff
@@ -504,8 +504,14 @@ public UCDeploySite getSite() {
...
     public void perform(final Run<?, ?> build, FilePath workspace, Launcher launcher, final TaskListener listener)
             throws AbortException, InterruptedException, IOException {
+        boolean pushFailedBuild = false;
+        pushFailedBuild = ((Push)getDelivery()).getPushFailedBuild();
         if (build.getResult() == Result.FAILURE || build.getResult() == Result.ABORTED) {
-            throw new AbortException("Skip artifacts upload to IBM UrbanCode Deploy - build failed or aborted.");
+            if(pushFailedBuild != true){
+                throw new AbortException("Skip artifacts upload to IBM UrbanCode Deploy - build failed or aborted.");
+            } else {
+                listener.getLogger().println("Continue to push even build is failed");
```
Comment (deepaannjohn) [inline src/main/java/com/urbancode/jenkins/plugins/ucdeploy/UCDeployPublisher.java:513]: Pushing
```suggestion
                listener.getLogger().println("Pushing Failed Build");   
```

File: src/main/java/com/urbancode/jenkins/plugins/ucdeploy/UCDeployPublisher.java [hunk 2]
```diff
@@ -504,8 +504,14 @@ public UCDeploySite getSite() {
     @Override
     public void perform(final Run<?, ?> build, FilePath workspace, Launcher launcher, final TaskListener listener)
             throws AbortException, InterruptedException, IOException {
+        boolean pushFailedBuild = false;
+        pushFailedBuild = ((Push)getDelivery()).getPushFailedBuild();
         if (build.getResult() == Result.FAILURE || build.getResult() == Result.ABORTED) {
-            throw new AbortException("Skip artifacts upload to IBM UrbanCode Deploy - build failed or aborted.");
+            if(pushFailedBuild != true){
```
Comment (deepaannjohn) [inline src/main/java/com/urbancode/jenkins/plugins/ucdeploy/UCDeployPublisher.java:510]:
```suggestion
            if(pushFailedBuild != true || build.getResult() == Result.ABORTED){
```

File: src/main/resources/com/urbancode/jenkins/plugins/ucdeploy/UCDeployPublisher/config.jelly [hunk 3]
```diff
@@ -86,6 +86,9 @@
             <f:entry field="pushIncremental" title="Incremental Version" help="${helpURL}/create/delivery/push/incremental.html">
               <f:checkbox/>
             </f:entry>
+            <f:entry field="pushFailedBuild" title="Push Failed build" help="${helpURL}/create/delivery/push/pushFailedBuild.html">
```
Comment (deepaannjohn) [inline src/main/resources/com/urbancode/jenkins/plugins/ucdeploy/UCDeployPublisher/config.jelly:89]:
```suggestion
            <f:entry field="pushFailedBuild" title="Push Failed Build" help="${helpURL}/create/delivery/push/pushFailedBuild.html">
```

**Review by deepaannjohn — CHANGES_REQUESTED** [review]

**Comment by AjeetGupta** [comment]
[Don't merge this code]

**Review by deepaannjohn — APPROVED** [review]

**Pull request merged**